### PR TITLE
pkp/pkp-lib#6222 Email discussion content to participants

### PR DESCRIPTION
**Pull request by Vitaliy-1**

### CODE REVIEW

File: classes/mail/traits/Discussion.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * @file mail/Discussion.php
```
Comment (NateWr) [inline classes/mail/traits/Discussion.php:4]:
```
@file classes/mail/traits/Discussion.php
```

File: classes/mail/traits/Discussion.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * @brief trait to support Discussion email template variables
+ */
+
+namespace PKP\mail\traits;
+
+trait Discussion
+{
+    protected static string $discussionSubject = 'discussionSubject';
+    protected static string $discussionContent = 'discussionContent';
```
Comment (NateWr) [inline classes/mail/traits/Discussion.php:21]: Let's call these just `subject` and `content`.

File: classes/mail/traits/Discussion.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        ]);
+    }
+
+    /**
+     * Add the title of the discussion and the content of the current note to the list of registered variables
+     */
+    protected static function addDiscussionDescription(array $variables): array
+    {
+        $variables[self::$discussionSubject] = __('emailTemplate.variable.discussion.subject');
+        $variables[self::$discussionContent] = __('emailTemplate.variable.discussion.message');
```
Comment (NateWr) [inline classes/mail/traits/Discussion.php:40]: Change the locale key to `.content` to match the the variable name.

File: controllers/grid/queries/QueriesGridHandler.php [hunk 4]
```diff
@@ -664,11 +664,15 @@ public function updateQuery($args, $request)
...
                 }
                 $submission = $this->getSubmission();
-                $mailable = $this->getStageMailable($request->getContext(), $submission);
-                $mailable->sender($currentUser);
-                $mailable->body($note->getData('contents'));
-                $mailable->subject($note->getData('title'));
-                $mailable->recipients([$user]);
+
+                $context = $request->getContext();
+                $mailable = $this->getStageMailable($context, $submission, $note->getData('title'), $note->getData('contents'));
```
Comment (NateWr) [inline controllers/grid/queries/QueriesGridHandler.php:669]: Is the user able to submit the new query form with an empty title? If so, can we add a test for that?

File: locale/en_US/emails.po [hunk 5]
```diff
@@ -266,6 +266,19 @@ msgstr ""
 msgid "emails.editorialReminder.description"
 msgstr "An automated email sent to editors with outstanding tasks"
 
+msgid "discussion.notification.subject"
+msgstr "{$dicussionSubject}"
```
Comment (NateWr) [inline locale/en_US/emails.po:270]: Typo here (`discu`). But I think this will be changed to just `{$subject}`...

File: locale/en_US/emails.po [hunk 6]
```diff
@@ -266,6 +266,19 @@ msgstr ""
...
+
+msgid "discussion.notification.body"
+msgstr ""
+"<p>{$discussionContent}</p>"
+"<hr>"
+"<p>You have a new notification from {$contextName}.</p>"
+"<p>Do not reply to this mail, please use this link: {$submissionUrl}</p>"
```
Comment (NateWr) [inline locale/en_US/emails.po:277]:
```
"<p>{$discussionContent}</p>"
"<hr>"
"<p>Reply to this message at <a href="{$submissionUrl}">{$contextName}</a>."
```

File: locale/en_US/emails.po [hunk 7]
```diff
@@ -266,6 +266,19 @@ msgstr ""
...
+
+msgid "discussion.notification.description"
+msgstr "The email is sent to registered users that have selected to have this type of notification emailed to them."
```
Comment (NateWr) [inline locale/en_US/emails.po:280]: > This email is sent when a user creates a new discussion or replies to an existing discussion.

File: locale/en_US/manager.po [hunk 8]
```diff
@@ -2652,6 +2652,12 @@ msgstr ""
 "The round of review this decision was taken in, if the decision is related "
 "to a review stage."
 
+msgid "emailTemplate.variable.discussion.subject"
+msgstr "The subject of the discussion"
+
+msgid "emailTemplate.variable.discussion.message"
+msgstr "The last message in discussion"
```
Comment (NateWr) [inline locale/en_US/manager.po:2659]: > The content of the new discussion or reply.

**Review by NateWr — CHANGES_REQUESTED** [review]

File: controllers/grid/queries/QueriesGridHandler.php (hunk 4 above)
Comment (Vitaliy-1) [reply controllers/grid/queries/QueriesGridHandler.php:669]: No, in the discussions/queries grid the subject is required. In participants grid the subject is retrieved from a template or by default from the NOTIFICATION template

**Pull request merged**